finesse/gui/temp_control.py

```diff
@@ -83,9 +84,13 @@ def _create_figure(self) -> None:
         self._figure_num_pts = int(
             TEMPERATURE_PLOT_TIME_RANGE / TEMPERATURE_MONITOR_POLL_INTERVAL
         )
-        t = [None] * self._figure_num_pts
-        hot_bb_temp = [None] * self._figure_num_pts
-        cold_bb_temp = [None] * self._figure_num_pts
+        t = [None] * self._figure_num_pts  # type: list[Optional[float]]
+        hot_bb_temp = [None] * self._figure_num_pts  # type: list[Optional[float]]
+        cold_bb_temp = [None] * self._figure_num_pts  # type: list[Optional[float]]
```

Comment:
I would do this in the following way:

```suggestion
        t: list[Optional[float]] = [None] * self._figure_num_pts
        hot_bb_temp: list[Optional[float]] = [None] * self._figure_num_pts
        cold_bb_temp: list[Optional[float]] = [None] * self._figure_num_pts
```

If done this way, you can also remove the noqa tag when importing Optional.